guides/source/asset_pipeline.md

```diff
@@ -868,218 +779,384 @@ each asset will have a different file name when it is changed. This way you
 don't have to ever manually invalidate any items in your cache. By using a
 different unique asset name instead, your users get the latest asset.
 
-Customizing the Pipeline
-------------------------
-
-### CSS Compression
-
-One of the options for compressing CSS is YUI. The [YUI CSS
-compressor](https://yui.github.io/yuicompressor/css.html) provides
-minification.
-
-The following line enables YUI compression, and requires the `yui-compressor`
-gem.
-
-```ruby
-config.assets.css_compressor = :yui
-```
-
-### JavaScript Compression
-
-Possible options for JavaScript compression are `:terser`, `:closure` and
-`:yui`. These require the use of the `terser`, `closure-compiler` or
-`yui-compressor` gems, respectively.
+Sprockets to Propshaft
+-------------------------------------
 
-Take the `terser` gem, for example.
-This gem wraps [Terser](https://github.com/terser/terser) (written for
-Node.js) in Ruby. It compresses your code by removing white space and comments,
-shortening local variable names, and performing other micro-optimizations such
-as changing `if` and `else` statements to ternary operators where possible.
+### Evolution of Asset Management Techniques
 
-The following line invokes `terser` for JavaScript compression.
-
-```ruby
-config.assets.js_compressor = :terser
-```
-
-NOTE: You will need an [ExecJS](https://github.com/rails/execjs#readme)
-supported runtime in order to use `terser`. If you are using macOS or
-Windows you have a JavaScript runtime installed in your operating system.
-
-NOTE: The JavaScript compression will also work for your JavaScript files when you are loading your assets through `importmap-rails` or `jsbundling-rails` gems.
-
-### GZipping your assets
-
-By default, gzipped version of compiled assets will be generated, along with
-the non-gzipped version of assets. Gzipped assets help reduce the transmission
-of data over the wire. You can configure this by setting the `gzip` flag.
-
-```ruby
-config.assets.gzip = false # disable gzipped assets generation
-```
+Within the last few years, the evolution of the web has led to significant
+changes that have influenced how assets are managed in web applications. These
+include:
 
-Refer to your web server's documentation for instructions on how to serve gzipped assets.
+1. **Browser Support**: Modern browsers have improved support for new features
+   and syntax, reducing the need for transpilation and polyfills.
+2. **HTTP/2**: The introduction of HTTP/2 has made it easier to serve multiple
+   files in parallel, reducing the need for bundling.
+3. **ES6+**: Modern JavaScript syntax (ES6+) is supported by most modern
+   browsers, reducing the need for transpilation and minification.
```

Comment:
```suggestion
   browsers, reducing the need for transpilation.
```

As noted further below, I didn't understand why the use of ES6 would reduce the need for minification. I thought minification became less necessary simply because the response is gzipped over the network (which is something done by `Rack::Deflater` unless I also misunderstood that part) and because nowadays there are even more efficient compression algorithms such as brotli.

Reply:
@sedubois you are correct, I miswrote here and they are unrelated. I've updated it accordingly.  Thank you for letting me know!